metrics/prometheus_test.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/containous/traefik/types"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+func TestPrometheus(t *testing.T) {
+	prometheusRegistry := RegisterPrometheus(&types.Prometheus{})
+	prometheusRegistry.ReqsCounter().With("service", "test", "code", strconv.Itoa(http.StatusOK), "method", http.MethodGet).Add(1)
+	prometheusRegistry.ReqsCounter().With("service", "test", "code", strconv.Itoa(http.StatusOK), "method", http.MethodGet).Add(1)
+	prometheusRegistry.ReqDurationHistogram().With("service", "test", "code", strconv.Itoa(http.StatusOK)).Observe(10000)
+	prometheusRegistry.ReqDurationHistogram().With("service", "test", "code", strconv.Itoa(http.StatusOK)).Observe(10000)
+	prometheusRegistry.RetriesCounter().With("service", "test").Add(1)
+
+	metricsFamilies, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		t.Fatalf("could not gather metrics families: %s", err)
+	}
+
+	tests := []struct {
+		name   string
+		labels map[string]string
+		assert func(*dto.MetricFamily)
+	}{
+		{
+			name: reqsTotalName,
+			labels: map[string]string{
+				"code":    "200",
+				"method":  http.MethodGet,
+				"service": "test",
+			},
+			assert: func(family *dto.MetricFamily) {
+				cv := uint(family.Metric[0].Counter.GetValue())
+				expectedCv := uint(2)
+				if cv != expectedCv {
+					t.Errorf("gathered metrics do not contain correct value for total requests, got %d expected %d", cv, expectedCv)
+				}
+			},
+		},
+		{
+			name: reqDurationName,
+			labels: map[string]string{
+				"service": "test",
+				"code":    "200",
+			},
+			assert: func(family *dto.MetricFamily) {
+				sc := family.Metric[0].Histogram.GetSampleCount()
+				expectedSc := uint64(2)
+				if sc != expectedSc {
+					t.Errorf("gathered metrics do not contain correct sample count for request duration, got %d expected %d", sc, expectedSc)
+				}
+			},
+		},
+		{
+			name: retriesTotalName,
+			labels: map[string]string{
+				"service": "test",
+			},
+			assert: func(family *dto.MetricFamily) {
+				cv := uint(family.Metric[0].Counter.GetValue())
+				expectedCv := uint(1)
+				if cv != expectedCv {
+					t.Errorf("gathered metrics do not contain correct value for total retries, got '%d' expected '%d'", cv, expectedCv)
```

Comment:
Doesn't sound right to me. When I change this it gets from:
`gathered metrics do not contain correct value for total requests, got 2 expected 23`
to
`gathered metrics do not contain correct value for total requests, got '\x02' expected '\x17'`.

Maybe though I should rather use floats for comparison and print messages with `%f`. WDYT?